specifyweb/stored_queries/execution.py

```diff
@@ -586,7 +588,7 @@ def build_query(session, collection, user, tableid, field_specs,
     query = QueryConstruct(
         collection=collection,
         objectformatter=ObjectFormatter(collection, user, replace_nulls),
-        query=session.query().distinct() if distinct else session.query(id_field),
+        query=session.query(func.group_concat(id_field, separator=',')) if distinct else session.query(id_field), # Also try id_field.distinct()
```

Comment:
` # Also try id_field.distinct`

No, don't lol. you can't just take a distinct on specific fields. Move it to group-by if you want to do that

Reply:
Here is a case of row row returned from a distinct query
`[ "20080,20080", "13265", "Characidae", "Acanthocharax sp.", null, "WGS 67-12", "03/17/1967",, "South America, Ecuador, Napo", 'Rio Aguarico, 400 yards South of "Munozlandia" at Santa Cecilia']`

In this case, where the two record IDs are the same, we don't want the user to be shown a potentially long dialog box of the same record ID.

> query=session.query(func.group_concat(id_field.distinct(), separator=',')) if distinct else session.query(id_field),

In SQLALCHEMY, this just makes the `GROUP_CONCAT(id, ',')` distinct, not any other field.  I can confirm this is the case, because the query result count does not change.

The previous row is now:
`[ "20080", "13265", "Characidae", "Acanthocharax sp.", null, "WGS 67-12", "03/17/1967",, "South America, Ecuador, Napo", 'Rio Aguarico, 400 yards South of "Munozlandia" at Santa Cecilia']`